### ISSUE TITLE
Request: Ability to assign mouse cursor movements to `input-remapper mouse` device

### ISSUE THREAD

**Issue by Tai-Mai** [sezanzeb/input-remapper#800]
The context is lengthy so please bear with me. 
I'm using a Trackball (Elecom Huge) and I'd like to use the ball to scroll by holding down a button that turns the ball into a scroll wheel. This is referred to as [on-button scrolling](https://wayland.freedesktop.org/libinput/doc/latest/scrolling.html#on-button-scrolling).
I currently do it with the middle click by doing the following commands (KDE, Wayland):
```zsh
$ qdbus org.kde.KWin /org/kde/KWin/InputDevice/event31 org.kde.KWin.InputDevice.scrollOnButtonDown true
$ qdbus org.kde.KWin /org/kde/KWin/InputDevice/event31 org.kde.KWin.InputDevice.scrollButton 274 
```
`event31` corresponds to the forwarded device and 274 is the kWin key code for the middle click. Mouse movements and any keys that I haven't remapped in input-remapper will belong to `event31`. I can verify that with the following command.
```zsh
$ libinput debug-events
```
The output will look something like this:
```
...
event31  POINTER_MOTION          +0.000s         0.00/  0.30 ( +0.00/ +1.00)
event31  POINTER_BUTTON          +13.617s      BTN_MIDDLE (274) pressed, seat count: 1
event31  POINTER_BUTTON          +13.794s      BTN_MIDDLE (274) released, seat count: 0
...
```
Now, on to my problem: I'd like to use another key than the middle click as the scroll button but currently, this only works when both the cursor movements and the scroll button belong to the same device. This means that I can't use any keys that I have remapped in input-remapper since they will belong to e.g. `event28` while the cursor movements still belong to `event31`. 

Is it possible to implement a feature that assigns the mouse movements to one of the targets as well?

[Logs](https://github.com/sezanzeb/input-remapper/files/13245143/input-remapper_debug.txt)

I also reported this as an [issue](https://bugreports.qt.io/browse/QTBUG-118745) to Qt in the hopes that they can fix this in `qdbus`.

**Comment by jonasBoss** [comment]
as long as the button is on your physical mouse you can do this all in input-remapper:
 1. Record an input with both the moues movement and the button press i.e. `Button MIDDLE + Y Up` (you need to move the mouse at a certain speed for it to recognize the movement)
 2. click on the `advanced` button and activate the "Use as analog" setting on the moues movement input.
![Screenshot_20231103_114001](https://github.com/sezanzeb/input-remapper/assets/31804124/dc4646c7-391c-4a35-b3a2-61bcd31c7149)
 3. In the Output Section go to the Analog Axis tab and set the Target to `mouse` and the Output Axis to `REL_WHEEL` for vertical or `REL_HWHEEL` for horizontal scrolling. 
![Screenshot_20231103_114724](https://github.com/sezanzeb/input-remapper/assets/31804124/72fd0660-6fa5-4e86-992a-52a642edf330)
 4. Play with the Deadzone, Gain and Expo settings to customize the scroll Behavior to your liking
 5. Of course you can do two mappings for both the vertical and horizontal scrolling and use the same button.

If your goal is to have the button on a different physical device i.e. your keyboard, this will not work as input-remapper can only combine inputs from the same physical device at the moment. But you can still use the Input-Remapper to send the mouse movement to the `mouse` target in exactly the same way as I described above.

**Comment by Tai-Mai** [comment]
Thank you! I didn't know this was possible in input-remapper.
Unfortunately, it's not a perfect replacement, unless I did something wrong. With the method I described above, I can do pixel-wise scrolling in wayland, so the scrolling is very smooth like with a touchpad instead of being stepped like a normal scroll wheel. With your method, it does work but it's a stepped scroll and not pixel-wise. I tried both the REL_WHEEL and REL_WHEEL_HI_RES targets.
Is there a way to make the scrolling smoother?

**Comment by jonasBoss** [comment]
There is nothing we can do on the Input-Remapper side for this. It is up to the the desktop environment and or the application to implement scrolling correctly. 

Using REL_WHEEL or REL_WHEEL_HI_RES  will always send both REL_WHEEL and REL_WHEEL_HI_RES events in a ratio of 1/120 to the mouse. Just as described in https://www.kernel.org/doc/html/latest/input/event-codes.html?highlight=evdev#ev-rel you can verify that by observing the input-remapper mouse with `evtest` or `libinput debug-events`

Basically input-remapper always sends smooth events with the REL_WHEEL_HI_RES event and for compatibility with software that still has not implemented hi res scrolling we also send REL_WHEEL events.

**Comment by jonasBoss** [comment]
according to https://kde.org/announcements/plasma/5/5.27.0/ _Plasma on Wayland also supports high resolution scroll wheels for smoother scrolling through long views._ So is this a plasma bug?

**Comment by jonasBoss** [comment]
> Is there a way to make the scrolling smoother?

Use your old approach and remap your button, as well as the mouse axis to the inputremapper-mouse

**Comment by Tai-Mai** [comment]
Thanks for looking into it! I initially reported this as an issue to Qt but they said it was the wrong place to do so. They pointed me towards reporting it to KDE instead.

> Use your old approach and remap your button, as well as the mouse axis to the inputremapper-mouse

This works, thank you! Feel free to close the issue.
